components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/client/IdentityProviderMgtServiceClient.java

```diff
@@ -113,20 +122,115 @@ public void updateResidentIdP(IdentityProvider identityProvider) throws Exceptio
      *
      * @return List of <code>FederatedIdentityProvider</code>. IdP names, primary IdP and home realm
      * identifiers of each IdP
-     * @throws Exception Error when getting list of Identity Providers
+     * @throws IdentityException Error when getting list of Identity Providers
      */
-    public List<IdentityProvider> getIdPs() throws Exception {
+    public IdentityProvider[] getIdPs() throws IdentityException {
+
         try {
-            IdentityProvider[] identityProviders = idPMgtStub.getAllIdPs();
-            if (identityProviders != null && identityProviders.length > 0) {
-                return Arrays.asList(identityProviders);
-            } else {
-                return new ArrayList<IdentityProvider>();
-            }
-        } catch (Exception e) {
+            return idPMgtStub.getAllIdPs();
+        } catch (RemoteException | IdentityProviderMgtServiceIdentityProviderManagementExceptionException e) {
             log.error("Error in retrieving the list of Identity Providers for a given tenant", e);
```

Comment:
Let's have log.error as follows, so that exception "e" and the error messages can differentiate easily.
log.error(" Error in retrieving the list of Identity Providers for a given tenant ", e);